index.js

```diff
@@ -153,12 +178,13 @@ TransformStreamDecode.prototype._flush = function(done) {
       return done(err);
     }
     if (output) {
-      that.push(output);
+      for (var i = 0; i < output.length; i++)
+        that.push(output[i]);
     }
     done();
-  });
+  }, !this.sync);
 };
 
-function decompressStream() {
-  return new TransformStreamDecode();
+function decompressStream(params) {
+  return new TransformStreamDecode(params);
```

Comment:
Mmh why did you add `params` to `TransformStreamDecode`? I don't see it being used.